server/jetstream_api.go

```diff
@@ -118,6 +118,12 @@ const (
 	JSApiMsgGet  = "$JS.API.STREAM.MSG.GET.*"
 	JSApiMsgGetT = "$JS.API.STREAM.MSG.GET.%s"
 
+	// JSDirectMsgGet is the template for non-api layer direct requests for a message by its stream sequence number or last by subject.
+	// Will return the message similar to how a consumer receives the message, no JSON processing.
+	// If the message can not be found we will use a status header of 404. If the stream does not exist the client will get a no-responders or timeout.
+	JSDirectMsgGet  = "$JS.DS.GET.*"
```

Comment:
How does this work with API prefixes? Currently if using prefixes, `$JS.API` is replaced with whatever prefix is provided.

Reply:
@derekcollison The current implementation doesn't correctly honor mux requests, the message is mapped to the subscription, but the subject is not including the inbox - go client I believe currently has an assumption that if there's only one mux it uses that, but that doesn't work if there's any other kind of request-reply going on with the client.

< PUB $JS.DS.GET.KV_a _INBOX.4APLG7K94VJXE3ZO5ZTI72.4APLG7K94VJXE3ZO5ZTIZI 26␍␊{"last_by_subj":"$KV.a.a"}␍␊
> HMSG $KV.a.a 1  49 54␍␊NATS/1.0␍␊Nats-Stream: KV_a␍␊Nats-Sequence: 1␍␊␍␊hello␍␊

Reply:
The current API for KV also provides the time when the entry was written which may or not be interesting to a client, the direct mechanism needs to provide this possibly in the headers.

Reply:
This is at the account level, not $SYS level so no API rewrite should be needed. Just like sending to a subject to be placed into a stream.

Reply:
Yes mux operations would need to be considered. We could make this mux safe and place original subject in header and reply with INBOX.

Reply:
I could add in Timestamp as well.

Reply:
if you add that, we can convert it and there's no difference to the current client contract.

Reply:
ok done.